Assets/BossRoom/Scripts/Shared/Game/State/GameStateBehaviour.cs

```diff
@@ -53,6 +55,22 @@ public virtual bool Persists
         /// </summary>
         private static GameObject s_ActiveStateGO;
 
+        protected DIScope m_Scope;
+
+        [SerializeField]
+        GameObject[] m_GameObjectsThatWillBeInjectedAutomatically;
+
+        protected virtual void Awake()
+        {
+            InitializeScope();
+            m_Scope.FinalizeScopeConstruction();
+            foreach (var autoInjectedGameObject in m_GameObjectsThatWillBeInjectedAutomatically)
+            {
+                m_Scope.InjectIn(autoInjectedGameObject);
+            }
+            DIScope.RootScope.InjectIn(this);
```

Comment:
That's redundant - scopes work as a tree, if the child doesn't have something - it looks in a parent scope.